Sources/FoundationEssentials/Platform.swift

```diff
@@ -172,31 +171,51 @@ extension Platform {
     
     static func name(forUID uid: uid_t) -> String? {
         withUserGroupBuffer(uid, passwd(), sizeProperty: Int32(_SC_GETPW_R_SIZE_MAX), operation: getpwuid_r) {
-            String(cString: $0.pw_name)
+            // Android's pw_name `char *`` is nullable, so always coerce to a nullable pointer
+            // in order to be compatible with Android.
+            let pw_name: UnsafeMutablePointer<CChar>? = $0.pw_name
+            return pw_name.flatMap { String(cString: $0) }
         }
     }
     
     static func fullName(forUID uid: uid_t) -> String? {
         withUserGroupBuffer(uid, passwd(), sizeProperty: Int32(_SC_GETPW_R_SIZE_MAX), operation: getpwuid_r) {
-            String(cString: $0.pw_gecos)
+#if os(Android) && (arch(i386) || arch(arm))
```

Comment:
Small comment: should we use `_pointerBitWidth(_32)` here just so we don't have to maintain a list of 32bit architectures if this ever changes?